### ISSUE TITLE
Orion returns illegal JSON in some situations

### ISSUE THREAD

**Issue by fgalan** [telefonicaid/fiware-orion#3280]
Creating this entity (note the `\n`):

```
$ curl localhost:1026/v2/entities -s -S -H 'Content-Type: application/json' -d @- <<EOF
{
  "id": "E",
  "type": "T",
  "A": {
    "value": "000FF\nABCDGC",
    "type": "Text"
  }
}
EOF
```

Stored at DB as:

```
{
	"_id" : {
		"id" : "E",
		"type" : "T",
		"servicePath" : "/"
	},
	"attrNames" : [
		"A"
	],
	"attrs" : {
		"A" : {
			"type" : "Text",
			"creDate" : 1535718069,
			"modDate" : 1535718069,
			"value" : "000FF\nABCDGC",
			"mdNames" : [ ]
		}
	},
	"creDate" : 1535718069,
	"modDate" : 1535718069,
	"lastCorrelator" : "57df79d6-ad18-11e8-9744-000c29173617"
}

```

You can retrieve it this way:

```
$ curl -s -S localhost:1026/v2/entities/E
{"id":"E","type":"T","A":{"type":"Text","value":"foo","metadata":{}},"A":{"type":"Text","value":"000FF
ABCDGC","metadata":{}}}
```

But note that is is illegal JSON.

```
$ curl -s -S localhost:1026/v2/entities/E | python -mjson.tool
Invalid control character at: line 1 column 103 (char 102)
```

Orion must not return never illegal JSON. The problem seems to be located in the rendering stage.

**Comment by fgalan** [comment]
Similar test, using a subscription:

```
$ curl -v localhost:1026/v2/subscriptions -s -S -H 'Content-Type: application/json' -d @- <<EOF
{
  "description": "A subscription\nwith a newline",
  "subject": {
    "entities": [
      {
        "id": "Room1",
        "type": "Room"
      }
    ]
  },
  "notification": {
    "http": {
      "url": "http://localhost:1028/accumulate"
    }
  }
}
EOF
```

In DB is:

```
{
	"_id" : ObjectId("5b8935d9e286793b32236883"),
	"expiration" : NumberLong("9223372036854775807"),
	"reference" : "http://localhost:1028/accumulate",
	"custom" : false,
	"throttling" : NumberLong(0),
	"servicePath" : "/#",
	"description" : "A subscription\nwith a newline",
	"status" : "active",
	"entities" : [
		{
			"id" : "Room1",
			"isPattern" : "false",
			"type" : "Room",
			"isTypePattern" : false
		}
	],
	"attrs" : [ ],
	"metadata" : [ ],
	"blacklist" : false,
	"conditions" : [ ],
	"expression" : {
		"q" : "",
		"mq" : "",
		"geometry" : "",
		"coords" : "",
		"georel" : ""
	},
	"format" : "normalized"
}
```

Rendered as legal JSON:

```
$ curl -s -S localhost:1026/v2/subscriptions/5b8935d9e286793b32236883 | python -mjson.tool
{
    "description": "A subscription\nwith a newline",
    "id": "5b8935d9e286793b32236883",
    "notification": {
        "attrs": [],
        "attrsFormat": "normalized",
        "http": {
            "url": "http://localhost:1028/accumulate"
        }
    },
    "status": "active",
    "subject": {
        "condition": {
            "attrs": []
        },
        "entities": [
            {
                "id": "Room1",
                "type": "Room"
            }
        ]
    }
}
```

This is due to subscriptions and entities are using different rendering logic. This remarks the importance of having only one way of rendering JSON along the code, so bugfixing can be done in just one place. This stresses the need of *unification and simplification* already described in https://github.com/telefonicaid/fiware-orion/issues/1298

**Comment by fgalan** [comment]
Fixed by PR #3285, .test and CNR added in PR https://github.com/telefonicaid/fiware-orion/pull/3291
